### PR TITLE
Move tools to a tools/ directory

### PR DESCRIPTION
**Pull request by whot**

### CODE REVIEW

**Comment by bluetech** [comment]
Thanks! LGTM, except that the `interactive-*` set probably also should be in tools/. (If you move them, make sure to `git grep test/interactive` as they are referred to by path name in a few places in the docs).

**Comment by whot** [comment]
@bluetech that's the lot now, I think. includes a renaming to prefix the tools with `xkbcommon-` (part of #151).

File: tools/print-compiled-keymap.c [hunk 1]
```diff
@@ -53,32 +57,64 @@ main(int argc, char *argv[])
...
+    ctx = xkb_context_new(XKB_CONTEXT_NO_FLAGS);
     if (!ctx) {
         fprintf(stderr, "Couldn't create xkb context\n");
-        goto err_out;
+        goto out;
     }
 
-    keymap = test_compile_file(ctx, keymap_path);
+    if (streq(keymap_path, "-")) {
+        FILE *tmp;
```
Comment (bluetech) [inline tools/print-compiled-keymap.c:67]: Heh stdio is pretty painful. I do think we need to read in a loop though, assuming the the stdin input is at most 4096 bytes is not great.

Comment (whot) [reply tools/print-compiled-keymap.c:67]: huh, I thought I had done that. Sorry, will fix.

**Comment by whot** [comment]
>  I think it'd be a bit nicer to duplicate (or open code) whatever functions are needed from there instead of including.

IMO it's best to split up so that `utils.h` is a generic utilities set like the [libinput `utils-foo.(c|h)` files](https://gitlab.freedesktop.org/libinput/libinput/-/tree/master/src). Then we can use it from anywhere, and the xkbcommon-specific utilities that require knowledge of headers can go into a separate file. but tbh, after a quick look at `utils.h` I don't see a function that needs to be in a more specific header.

**Comment by whot** [comment]
> rmlvo-to-kccgst.c includes xkbcomp/xkbcomp-priv.h and xkbcomp/rules.h. This perhaps indicates some missing public API, though we got by without exposing kccgst at all, so maybe we should just ignore this one, and keep the private linking.

had a look at this and yeah, it would require exposing KcCGST as public API, which is something to avoid. So I'd say let's keep it at this.

the `stdin` handling is fixed (I think, it's getting late here :) in the last set of commits, only the top commit changed.

**Comment by whot** [comment]
Thanks! merging

**Pull request merged**